app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,54 @@
+module Api
+  module V1
+    class ReservationsController < ApplicationController
+      load_and_authorize_resource
+      before_action :set_reservation, only: %i[show update destroy]
+
+      def index
+        @reservations = current_api_v1_user.reservations
+        render json: @reservations
+      end
```

Comment:
When rendering `reservations`, it is not only the IDs for car and reservations that will be required on the front end but also the data for the associated cars. I would suggest eager loading the associated cars with the reservations and rendering them too.
```suggestion
      def index
        @reservations = current_api_v1_user.reservations.includes(:car)
        render json: @reservations, include: :car
      end
```